### DISCUSSION TITLE
Trying to upload multiple files fails.

### DISCUSSION THREAD

**Discussion by apoorv569** [nextcloud/all-in-one#1555] [Questions] [answered]
I am trying to upload <=500 photos at once and the process fails and in the `VM` running the `Nextcloud AIO` in the `TTY` I see this error,

![2022-12-20_00-51](https://user-images.githubusercontent.com/53279454/208503544-2d2e26e5-8005-4db0-9dc0-ca4c44aa1d66.png)

Sometimes it is about `Imaginary` and sometimes it is about `Java`?

Is it possible to increase memory for these processes?

**Comment by szaimen** [comment]
Hm... You could try to increase the RAM of the VM. Apart from that you might try to disable imaginary and see if that works better.

**Reply by apoorv569** [reply]
> Hm... You could try to increase the RAM of the VM. Apart from that you might try to disable imaginary and see if that works better.

I have tried increasing RAM to 8 and even 16 GB but the problem remains. I noticed while upload the RAM gets fully used no matter if I give it 8 or 16 and this is just for 500 something files what if I try to upload thousands of files at one. There should be some optimization that can be done here. BTW I can't really give the VM 16GB I just tried for once, max I can give 8GB.

BTW what is imaginary for?

**Reply by szaimen** [reply]
Imaginary is used for preview generation. However the Nextcloud container is able to do it also itself. That is why I recommended to disable it and check if it makes it work.

**Reply by apoorv569** [reply]
> Imaginary is used for preview generation. However the Nextcloud container is able to do it also itself. That is why I recommended to disable it and check if it makes it work.

I see. I disabled Imaginary and everything else (clamav, collabora, fulltextsearch and all) except nextcloud itself and redis and tried uploading those 500 files again.. At first it failed but not with the same error as before I just saw this,

![2022-12-20_14-52](https://user-images.githubusercontent.com/53279454/208646329-e3e7dc4d-7f9e-4881-a915-d8df41d66c92.png)

But it kept uploading so I waited but eventually it failed completely. I restarted and tried one more time and second time it worked had couple of these errors show up but it did finish.

Also I noticed opening the Photos directory which contains over 10-15k photos is very slow with or without imaginary.

**Reply by szaimen** [reply]
> I see. I disabled Imaginary and everything else (clamav, collabora, fulltextsearch and all) except nextcloud itself and redis and tried uploading those 500 files again.. At first it failed but not with the same error as before I just saw this,
> 
> ![2022-12-20_14-52](https://user-images.githubusercontent.com/53279454/208646329-e3e7dc4d-7f9e-4881-a915-d8df41d66c92.png)
> 
> But it kept uploading so I waited but eventually it failed completely. I restarted and tried one more time and second time it worked had couple of these errors show up but it did finish.

All right!

> Also I noticed opening the Photos directory which contains over 10-15k photos is very slow with or without imaginary.

Yes, this is unfortunately a known issue of Nextcloud. Putting this many files into one folder is not scaling well.

**Reply by apoorv569** [reply]
I see. Is there some app that can organize all these files in to sub-folders by like year or month perhaps? As doing this manually with this many files is not feasible.

**Reply by szaimen** [reply]
Maybe use the [desktop client](https://github.com/nextcloud/desktop) for this?

**Reply by apoorv569** [reply]
> Maybe use the [desktop client](https://github.com/nextcloud/desktop) for this?

I found couple of scripts online on `github` that could help sort these photos by year/month, perhaps this could help me organize the files a little better.

For anyone that comes across this discussion with similar issue, here are the links to scripts that I found,

1. https://github.com/andrewning/sortphotos
2. https://github.com/ivandokov/phockup
3. https://github.com/johnkristijan/photo-organizer-by-year-and-month

**Comment by szaimen** [comment, marked as answer]
Lets mark your answer as the solution:

> > Maybe use the [desktop client](https://github.com/nextcloud/desktop) for this?
> 
> I found couple of scripts online on `github` that could help sort these photos by year/month, perhaps this could help me organize the files a little better.
> 
> For anyone that comes across this discussion with similar issue, here are the links to scripts that I found,
> 
> 1. https://github.com/andrewning/sortphotos
> 2. https://github.com/ivandokov/phockup
> 3. https://github.com/johnkristijan/photo-organizer-by-year-and-month
